tests/test_browser.py

```diff
@@ -5103,6 +5103,17 @@ def test_assert_failure(self):
   def test_full_js_library_strict(self):
     self.btest_exit(test_file('hello_world.c'), args=['-sINCLUDE_FULL_LIBRARY', '-sSTRICT_JS'])
 
+  # Tests audio worklets
+  @requires_threads
+  @requires_sound_hardware
+  def test_audio_worklet(self):
+    self.btest(path_from_root('tests', 'audioworklet', 'audioworklet.cpp'),
+               expected='1',
+               args=['-s', 'USE_PTHREADS=1', '-s', 'MODULARIZE=1', '-s',
```

Comment:
Is `MODULARIZE` needed to use worklets?

Reply:
Yes, unfortunately it's needed because AudioWorkletGlobalScope doesn't let us fetch/import scripts from the worklet context which is what regular pthreads do (they load the `.worker.js` which then does `importScript` in the `load` message from the main thread). 

In the worklet case we need to use `audioCtx.addModule` on the main thread to load both the `.worker.js` and the main js and then when they're both loaded and the `load` message is sent over we can just skip the import step and run the module function. 

If we don't MODULARIZE then then the main js executes before all the things are set up in the `load` message coming from the main thread (the `wasmModule` etc). For the regular pthread case this is not a problem, because `importScripts` happens in `load` only after this has been set up already.

Reply:
I actually found a way to make it work without MODULARIZE, so this is resolved now.